include/clasp/core/object.h

```diff
@@ -511,15 +511,13 @@ inline CL_DEFUN bool cl__eql(T_sp x, T_sp y) {
   if (x.fixnump()) {
     return x.raw_() == y.raw_();
   } else if (x.single_floatp()) {
-    if (y.single_floatp()) {
-      return gc::tagged_single_float_masked(x.raw_()) == gc::tagged_single_float_masked(y.raw_());
-    }
-    return false;
+    if (!y.single_floatp())
+      return false;
+    single_float_t xf = x.unsafe_single_float();
+    single_float_t yf = y.unsafe_single_float();
+    return xf == yf && std::signbit(xf) == std::signbit(yf);
```

Comment:
probably worth mentioning the point of this (signed zero comparisons) in a comment, since it's a bit inscrutable